test/controllers/ConfirmationPageControllerSpec.scala

```diff
@@ -16,38 +16,60 @@
 
 package controllers
 
-import config.AppConfig
-import connectors._
-import models.*
+import connectors.{CustomsDataStoreConnector, CustomsFinancialsApiConnector}
+import models.CashTransactionDates
 import org.mockito.ArgumentMatchers.any
 import org.mockito.Mockito.when
 import play.api.Application
+import play.api.i18n.{Messages, MessagesApi}
 import play.api.inject.bind
 import play.api.mvc.AnyContentAsEmpty
 import play.api.test.FakeRequest
 import play.api.test.Helpers.*
 import repositories.RequestedTransactionsCache
+import uk.gov.hmrc.auth.core.retrieve.Email
 import utils.SpecBase
 
 import java.time.LocalDate
 import scala.concurrent.Future
 
 class ConfirmationPageControllerSpec extends SpecBase {
 
-  "call page load returns valid response" in new Setup {
+  "calling page load returns valid response with email address" in new Setup {
     when(mockRequestedTransactionsCache.get(any)).thenReturn(Future.successful(Some(cashDates)))
+    when(mockCustomsDataStoreConnector.getEmail(any)(any)).thenReturn(Future.successful(Right(Email(email))))
 
     val request: FakeRequest[AnyContentAsEmpty.type] =
       fakeRequest(GET, routes.ConfirmationPageController.onPageLoad().url)
 
     running(app) {
       val result = route(app, request).value
       status(result) mustBe OK
+
+      val content = contentAsString(result)
+      content must include(messages("cf.cash-account.transactions.confirmation.email", email))
+    }
+  }
+
+  "calling page load returns valid response without email address" in new Setup {
+    when(mockRequestedTransactionsCache.get(any)).thenReturn(Future.successful(Some(cashDates)))
+    when(mockCustomsDataStoreConnector.getEmail(any)(any)).thenReturn(Future.successful(Right(Email(emptyString))))
```

Comment:
It should be returning either Left(UndeliverableEmail) or Left(UnverifiedEmail) when email value is absent